### ISSUE TITLE
Battlegroup Problems

### ISSUE THREAD

**Issue by gelir95** [Wyqer/kp_liberation#367]
### Basic Information
**Mission version: 0.963
**Map used: Lythium
**Mods used: ACE, ACRE, RHS, Project Opfor
**Dedicated or local game: Dedicated 
**Source of the mission file (URL):**

### Individual Things
**Did you edit anything within the mission files?**

Yes but nothing inside the spawn logic.

**If yes, what did you edit exactly?**

### Short Issue Description

Battlegroup infantry is not using the transport vehicle and walking instead.

### Expected Behaviour

Infantry using the vehicle.

### Steps to Reproduce the Issue

Wait for a battlegroup to spawn.

### Screenshots, Serverlog or any other helpful Information

https://i.imgur.com/R7XDGql.png (infantry and vehicles on the south)

Same behavior with Liberation 0.963 with a not edited mission file.

**Comment by Wyqer** [comment]
I don't see any transport vehicle like an Ural truck or similar which should transport the infantry troops.

Anything changed in the enemy preset concerning troop transport vehicles?

References in the code:
https://github.com/Wyqer/kp_liberation/blob/master/Missionframework/scripts/server/battlegroup/spawn_battlegroup.sqf#L41
https://github.com/Wyqer/kp_liberation/blob/master/Missionframework/scripts/server/ai/troup_transport.sqf

**Comment by gelir95** [comment]
Localy reproduced the behavior with https://github.com/Wyqer/kp_liberation/blob/master/Missionframework/presets/opfor/takistan.sqf active.

**Comment by Wyqer** [comment]
Replace the two mentioned files above in your missionfile with the ones I'll send you via Discord.
If you encounter it again, attach a serverlog pls.

**Comment by Wyqer** [comment]
I've checked your issue description with tests on my side.
The troop transport behaviour for battlegroups work as intended.
Maybe you "look too fast" and don't look "long enough" what's going on, so you think it's not working.

* Battlegroup spawns with target.
* If there is a vehicle which is a "transport vehicle" it will start the transport AI script.
* The transport script waits until the "transport vehicle" is under a distance of 1000m to the target.
* Then the "cargo infantry" spawns in and the transport will unload the Infantry (heli land, etc.)
* After unloading the transport vehicle and the "cargo infantry" get SAD waypoints around the target area.

Tested and works without any issues with the official code.
